prbt_hardware_support/test/unittests/unittest_sto_modbus_adapter.cpp

```diff
@@ -131,17 +141,22 @@ TEST_F(PilzStoModbusAdapterTest, testSetup)
 }
 
 /**
- * @brief Test an expection is thrown if there is no disable service to disable the drives
+ * @brief Test constructor with delayed halt service
+ *
+ * Expected: Constructor blocks until the halt service is available.
  */
 TEST_F(PilzStoModbusAdapterTest, testSetupNoDisableService)
 {
   manipulator_.advertiseHoldService(nh_, HOLD_SERVICE_T);
   manipulator_.advertiseUnholdService(nh_, UNHOLD_SERVICE_T);
   manipulator_.advertiseRecoverService(nh_, RECOVER_SERVICE_T);
 
-  EXPECT_THROW(PilzStoModbusAdapterNode adapter_node(nh_), PilzStoModbusAdapterNodeException);
+  auto t = asyncConstructor();
+  ros::Duration(5.5).sleep();
```

Comment:
Maybe add a small comment explaining the choice of the duration.

Reply:
Yep, good point.